internal/app/upstream/client.go

```diff
@@ -180,7 +175,7 @@ func send(
 	logger log.Logger,
 	cancelFunc context.CancelFunc,
 	request transport.Request,
```

Comment:
Do you think we can omit `request` from send since it's been abstracted? I know the logger uses it below, but I wonder if we can also move that to the abstracted layer (stream.go) instead.

Reply:
Chatted offline, request version being logged is still incremented.

Reply:
since its a private function and we use it for logging, i would leave it there. The node id is not important to log here, so i will remove it in a next PR. the other 2 values make sense to log.